### ISSUE TITLE
Istio envoy proxy side car pass control to application container before it is fully ready

### ISSUE THREAD

**Issue by jshum2479** [istio/istio#29766]
(**NOTE**: This is used to report product bugs:
  To report a security vulnerability, please visit <https://istio.io/about/security-vulnerabilities>
  To ask questions about how to use Istio, please visit <https://discuss.istio.io>)

**Bug description**

Hi,  I have a question on how the side-car injection works.  We found random failures when installing our application in the testing environment, the first thing the application does is to contact the api server to check the k8s version.  We have the istio auto injection enabled, but intermittently it failed to contact the api-server resulting a
java.net.ConnectionException

From the proxy side car log and our container log,  it shows the envoy side car is not ready before we start the rest call to the server .. usually within a second.  How does the side car injection works?  Is it doing some async initializations and then immediate pass the flow to the app container?  We start seeing this behavior after upgrading from 1.5.4 to 1.7.x (edited) 
**Affected product area (please put an X in all that apply)**

[ ] Docs
[ ] Installation
[ X] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Expected behavior**
The envoy proxy should pass control to application container only when it is ready.
**Steps to reproduce the bug**
The first step the application container use rest call to the api-server to check the version of k8s.
**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)
1.7.3 
**How was Istio installed?**

istioctl install --set profile=demo -y

**Environment where the bug was observed (cloud vendor, OS, etc)**
k8s 1.16
Additionally, please consider running `istioctl bug-report` and attach the generated cluster-state tarball to this issue.
Refer [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) for more details.

2020-12-22T21:58:10.761513Z	info	Envoy proxy is ready

{"timestamp":"12-22-2020T21:58:07.922+0000","thread":1,"fiber":"","namespace":"","domainUID":"","level":"WARNING","class":"oracle.kubernetes.operator.helpers.HealthCheckHelper","method":"performK8sVersionCheck","timeInMillis":1608674287922,"message":"Failure reading Kubernetes version.  Exception thrown: {0}","exception":"\nio.kubernetes.client.openapi.ApiException: java.net.ConnectException: Failed to connect to /10.96.0.1:443

**Comment by howardjohn** [comment]
This is a duplicate of https://github.com/istio/istio/issues/11130, please see that issue for details and workarounds and long term fix plan. Thanks!
